synapse/handlers/identity.py

```diff
@@ -114,10 +114,7 @@ def threepid_from_creds(self, id_server, creds):
 
         query_params = {"sid": session_id, "client_secret": client_secret}
 
-        url = "https://%s%s" % (
-            id_server,
-            "/_matrix/identity/api/v1/3pid/getValidated3pid",
-        )
+        url = "%s%s" % (id_server, "/_matrix/identity/api/v1/3pid/getValidated3pid")
```

Comment:
1. if we're assuming `id_server` has to include a protocol now, please update the docstring, and chase that up the call stack.

2. if the id server comes from the `creds` dict: presumably it will be lacking a protocol?

3. I'd write this as `url = id_server + "/..."`

Reply:
Mm, yes `id_server` shouldn't need to be `None` anymore, so I'll remove `id_server` from the creds dict.